agent/connect/ca/provider_vault.go

```diff
@@ -24,41 +24,51 @@ var ErrBackendNotInitialized = fmt.Errorf("backend not initialized")
 type VaultProvider struct {
 	config    *structs.VaultCAProviderConfig
 	client    *vaultapi.Client
+	isRoot    bool
 	clusterId string
 }
 
 // NewVaultProvider returns a vault provider with its root and intermediate PKI
 // backends mounted and initialized. If the root backend is not set up already,
 // it will be mounted/generated as needed, but any existing state will not be
 // overwritten.
-func NewVaultProvider(rawConfig map[string]interface{}, clusterId string) (*VaultProvider, error) {
-	conf, err := ParseVaultCAConfig(rawConfig)
+func NewVaultProvider() *VaultProvider {
+	return &VaultProvider{}
+}
```

Comment:
Same comment as before - this doesn't seem to add value any more and the comment is out of date.